src/main/scala/leon/utils/package.scala

```diff
@@ -26,5 +26,17 @@ package object utils {
     }
     v2
   }
-  
+
+  /** generate a two-digit (0-f) hex string for the given byte. */
+  def toByteHex(b: Byte): String = {
```

Comment:
@regb I think for now it would be over-engineering things, but if in the future more kinds of integer are supported then scala-smtlib's `Hexadecimal` could be refactored with a generalisation of `toByteHex` for any integer length (or maybe limited to length <= 32).